File: src/com/mrpowergamerbr/loritta/commands/vanilla/fun/TranslateCommand.java
package com.mrpowergamerbr.loritta.commands.vanilla.fun;

import java.util.Arrays;
import java.util.List;

import com.mrpowergamerbr.loritta.Loritta;
import com.mrpowergamerbr.loritta.commands.CommandBase;
import com.mrpowergamerbr.loritta.commands.CommandCategory;
import com.mrpowergamerbr.loritta.commands.CommandContext;
import com.mrpowergamerbr.loritta.utils.translate.GoogleTranslateUtils;

public class TranslateCommand extends CommandBase {
	@Override
	public String getLabel() {
		return "traduzir";
	}

	@Override
	public String getDescription() {
		return "Traduz uma frase para outra linguagem";
	}

	@Override
	public String getUsage() {
		return "língua texto";
	}

	@Override
	public List<String> getExample() {
		return Arrays.asList("pt Hello World!");
	}

	@Override
	public CommandCategory getCategory() {
		return CommandCategory.FUN;
	}

	@Override
	public void run(CommandContext context) {
		if (context.getArgs().length >= 2) {	
			String strLang = context.getArgs()[0];
			context.getArgs()[0] = ""; // Super workaround
			String text = String.join(" ",context.getArgs());

			try {
				String translatedText = GoogleTranslateUtils.translate(text, "auto", strLang);

				context.sendMessage(context.getAsMention(true) + translatedText);
			} catch (Exception e) {
				e.printStackTrace();
			}
		} else {
			context.explain();
		}
	}
}

File: src/main/java/com/mrpowergamerbr/loritta/commands/vanilla/utils/TranslateCommand.kt
package com.mrpowergamerbr.loritta.commands.vanilla.utils

import com.mrpowergamerbr.loritta.commands.AbstractCommand
import net.perfectdreams.loritta.api.commands.CommandCategory
import com.mrpowergamerbr.loritta.commands.CommandContext
import com.mrpowergamerbr.loritta.utils.escapeMentions
import com.mrpowergamerbr.loritta.utils.locale.LegacyBaseLocale
import com.mrpowergamerbr.loritta.utils.translate.GoogleTranslateUtils

class TranslateCommand : AbstractCommand("traduzir", listOf("translate"), CommandCategory.UTILS) {
	override fun getDescription(locale: LegacyBaseLocale): String {
		return locale["TRANSLATE_DESCRIPTION"]
	}

	override fun getUsage(): String {
		return "língua texto"
	}

	override fun getExamples(): List<String> {
		return listOf("pt Hello World!")
	}

	override suspend fun run(context: CommandContext,locale: LegacyBaseLocale) {
		if (context.args.size >= 2) {
			val strLang = context.args[0]
			context.args[0] = "" // Super workaround
			val text = context.args.joinToString(" ")

			try {
				val translatedText = GoogleTranslateUtils.translate(text, "auto", strLang)
				context.sendMessage(context.getAsMention(true) + translatedText!!.escapeMentions())
			} catch (e: Exception) {
				e.printStackTrace()
			}

		} else {
			context.explain()
		}
	}
}